[snapshot] Add `suppress_xcode_output` option

rogerluan opened the pull request:
🔑 

### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context

Thanks to the `You can do this` label, this PR resolves #16792 🚀 

### Description

This change is quite simple actually, it simply adds a `suppress_xcode_output` to `snapshot` action, similar to the analogous argument in `scan` and `gym`.

### Testing Steps

`bundle exec fastlane snapshot --suppress_xcode_output=true`

rogerluan committed:
Write specs for snapshot suppress_xcode_output parameter.

rogerluan committed:
Add suppress_xcode_output parameter to snapshot.

rogerluan committed:
Add missing spec to validate snapshot disable_xcpretty parameter.

rogerluan commented:
I can't see what failed there ☝️ the tests pass locally, I may need some help 😬

joshdholtz commented on snapshot/lib/snapshot/test_command_generator_xcode_8.rb:
```diff
@@ -24,7 +24,10 @@ def generate(device_type: nil, language: nil, locale: nil)
         log_path = xcodebuild_log_path(device_type: device_type, language: language, locale: locale)
-        return ["| tee #{log_path.shellescape} | xcpretty #{Snapshot.config[:xcpretty_args]}"]
+        pipe = "| tee #{log_path.shellescape} "
+        pipe << "| xcpretty #{Snapshot.config[:xcpretty_args]}"
+        pipe << "> /dev/null" if Snapshot.config[:suppress_xcode_output]
```
Do we need a space before the `>`?

rogerluan replied:
I had initially written it with a space, and while testing I noticed it'd print out e.g. `…| xcpretty  > /dev/null` (double spaces), so I removed the leading space here.
But now, revisiting this, there's also the `xcpretty_args` case, which needs a space.

Do we care about the command style if it contains double spaces? It's just a mere styling issue cuz obv it works with multiple extra whitespaces :) If we do care about making the command look pretty, I gotta check how to write this neatly in ruby 😇

rogerluan replied:
```suggestion
        pipe << "| xcpretty #{Snapshot.config[:xcpretty_args]}"
        pipe << " > /dev/null" if Snapshot.config[:suppress_xcode_output]
```

joshdholtz commented on snapshot/lib/snapshot/test_command_generator.rb:
```diff
@@ -33,8 +33,9 @@ def pipe(log_path: nil)
         xcpretty << "--no-color" if Helper.colors_disabled?
-
-        return pipe << "| #{xcpretty}"
+        pipe << "| #{xcpretty}"
+        pipe << "> /dev/null" if Snapshot.config[:suppress_xcode_output]
```
Do we need a space before the `>`?

rogerluan replied:
```suggestion
        pipe << "| #{xcpretty}"
        pipe << " > /dev/null" if Snapshot.config[:suppress_xcode_output]
```

joshdholtz reviewed (changes requested):
This looks great! Thanks for tackling this 😊 Just a question on if a space is needed before the `>` 😇

rogerluan committed:
Add extra space before piping the output.

rogerluan commented:
Thanks for bringing that up @joshdholtz ! Check my comment above on the rationale 🙏 For now, I added the extra space in both occurrences that you raised. If you agree, it's good to be merged 💪

joshdholtz commented on snapshot/lib/snapshot/test_command_generator.rb:
```diff
@@ -33,8 +33,9 @@ def pipe(log_path: nil)
-
-        return pipe << "| #{xcpretty}"
+        pipe << "| #{xcpretty}"
+        pipe << " > /dev/null" if Snapshot.config[:suppress_xcode_output]
```
```suggestion
        pipe << "> /dev/null" if Snapshot.config[:suppress_xcode_output]
```

😱 My bad! I didn't realize this pipe was going into an array. This array later on gets a `.join(' ')` which adds the space. Sorry about that!

rogerluan replied:
Oh, I missed that. Thanks for catching it!

joshdholtz commented on snapshot/lib/snapshot/test_command_generator_xcode_8.rb:
```diff
@@ -24,7 +24,10 @@ def generate(device_type: nil, language: nil, locale: nil)
       def pipe(device_type, language, locale)
         log_path = xcodebuild_log_path(device_type: device_type, language: language, locale: locale)
-        return ["| tee #{log_path.shellescape} | xcpretty #{Snapshot.config[:xcpretty_args]}"]
+        pipe = "| tee #{log_path.shellescape} "
+        pipe << "| xcpretty #{Snapshot.config[:xcpretty_args]}"
+        pipe << " > /dev/null" if Snapshot.config[:suppress_xcode_output]
+        return [pipe]
```
```suggestion
        pipe = ["| tee #{log_path.shellescape}"]
        pipe << "| xcpretty #{Snapshot.config[:xcpretty_args]}"
        pipe << "> /dev/null" if Snapshot.config[:suppress_xcode_output]
        return pipe
```

This one I think is what got me thinking weird about the other pipe 🤷‍♂️ This suggested changed makes the initial pipe an array instead of a string. The space in between will happen in the `.join(' ')` later on 😊

rogerluan replied:
Oooh sweet. This makes more sense! I tried keeping the original logic as much as possible but I see the difference now 😇

joshdholtz reviewed (changes requested):
Hey, sorry! I think I got some things confused on last review. I added two more suggestions should make this all gucci with spaces 😇

rogerluan committed:
Refactor pipes logic fix spacing between arguments.

The array of strings that build up the pipe will perform the `.join(" ")` of its elements.

Co-authored-by: Josh Holtz <josh@rokkincat.com>

rogerluan commented:
Looks g2g now 😄

joshdholtz reviewed (approved):
This looks great!! Thank you so much for another awesome contribution ❤️ Really appreciate it!
